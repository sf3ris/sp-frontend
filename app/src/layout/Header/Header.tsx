import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUserCircle } from '@fortawesome/free-regular-svg-icons';
import { useHistory } from 'react-router-dom';
import { faArrowRight, faSignOutAlt } from '@fortawesome/free-solid-svg-icons';
import { useDispatch } from 'react-redux';
import { logout } from '../../shared/user/slices/userSlices';


const Header : React.FC<{}> = props => {

    const history = useHistory();

    const dispatch = useDispatch();

    const onUserClick = () => {

        history.push( '/profile');

    }

    const onLogout = () => {

        dispatch( logout(null) );

    }

    return (

        <section className="search-and-user">
            <form>
                <input type="search" value="Page Title" disabled></input>
            </form>
            <div className="admin-profile">

                <span className="greeting">Administrator</span>
                <div onClick={onUserClick} className="notifications">
                    <span className="badge">1</span>
                    <FontAwesomeIcon icon={faUserCircle} />
                </div>

                <div onClick={onUserClick} style={{marginLeft:'2vw'}}>
                    <FontAwesomeIcon onClick={onLogout} icon={faSignOutAlt} />
                </div>

            </div>
        </section>

    )

}

export default Header;