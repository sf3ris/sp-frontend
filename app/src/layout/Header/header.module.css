.header {
    padding: 0; 
    background-color:#dfdfdf;
    height:3.75rem;
    display:flex;
    flex-direction: row;
    box-shadow: 0 .15rem 1.75rem 0 rgba(58,59,69,.15) !important
}

.container {
    margin-top: auto;
    margin-bottom: auto;
    padding-left: 3rem;
    display: flex;
    flex-direction: row;
}

.containerItem {
    padding: 1rem;
}

.container:hover {
    cursor: pointer;
}