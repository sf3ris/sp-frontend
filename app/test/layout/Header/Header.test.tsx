import React from 'react';
import { shallow,configure } from 'enzyme';
import Header from '../../../src/layout/Header/Header';
import Adapter from 'enzyme-adapter-react-16';
import * as ReactRedux from 'react-redux';

const useDispatchMock : jest.SpyInstance = jest.spyOn(ReactRedux,'useDispatch');

configure({adapter: new Adapter()});

describe('Header layout component test', () => {

    it("Should match snapshot", () => {

        useDispatchMock.mockImplementation(()=>null);

        const wrapper = shallow( <Header /> );

        expect(wrapper.debug()).toMatchSnapshot();

    })
    
})

