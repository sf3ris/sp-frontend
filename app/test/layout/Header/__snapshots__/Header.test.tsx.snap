// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header layout component test Should match snapshot 1`] = `
"<section className=\\"search-and-user\\">
  <form>
    <input type=\\"search\\" value=\\"Page Title\\" disabled={true} />
  </form>
  <div className=\\"admin-profile\\">
    <span className=\\"greeting\\">
      Administrator
    </span>
    <div onClick={[Function: onUserClick]} className=\\"notifications\\">
      <span className=\\"badge\\">
        1
      </span>
      <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false} />
    </div>
    <div onClick={[Function: onUserClick]} style={{...}}>
      <FontAwesomeIcon onClick={[Function: onLogout]} icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false} />
    </div>
  </div>
</section>"
`;
